eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1327 @@
+/* Copyright 2018 Endless Mobile, Inc. */
+
+#include <string.h>
+#include <stdlib.h>
+
+#include <endless/endless.h>
+#include <json-glib/json-glib.h>
+#include <mustache.h>
+#include <libintl.h>
+
+#include "eknr-boxed-set-info.h"
+#include "eknr-errors.h"
+#include "eknr-renderer.h"
+
+#define _(String) gettext (String)
+#define gettext_noop(String) String
+#define N_(String) gettext_noop (String)
+
+/**
+ * SECTION:renderer
+ * @title: Renderer
+ * @short_description: A post-processing renderer for HTML article content
+ *
+ * The renderer is responsible for adding some final postprocessing to
+ * articles on the client side before it is displayed to the user. What
+ * postprocessing happens depends on the article's source.
+ */
+struct _EknrRenderer
+{
+  GObject parent_instance;
+};
+
+typedef struct _EknrRendererPrivate
+{
+  GHashTable *cache;
+} EknrRendererPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (EknrRenderer,
+                            eknr_renderer,
+                            G_TYPE_OBJECT)
+
+enum {
+  PROP_0,
+  NPROPS
+};
+
+static GParamSpec *eknr_renderer_props [NPROPS] = { NULL, };
```

Comment:
This and the above enum are now unused

Reply:
Side note: Now that we don't have properties, I was thinking that we could make the renderer a pure function, though it is kinda handy to have a cache of compiled templates around...